### ISSUE TITLE
Quit VSCode when last window is closed on MacOS

### ISSUE THREAD

**Issue by javierguzman** [microsoft/vscode#176161]
Hello all,

I just come from the issue #25811 related to MacOS and I think it would be a great feature to have something similar to what iTerm2 has. This is quitting the application if the last window has been closed. Right now if I click to the red button it just closes VSCode window but the app stays open in the background (I can see it in the dock) and I have to close using cmd + Q. I would like to have an option to decide whether to keep VSCode open or not if the last window is closed.

Regards,
Javier

**Comment by tmadaj** [comment]
Yes please. I recently switched to macOS and absolutely hate this behaviour in the majority of apps.

**Comment by flying-sheep** [comment]
I like keeping apps around that do background work (messengers or so), but video players, terminals, and editors have no business staying around when there’s nothing visible.

**Comment by suanfalalala** [comment]
support,I hate this behaviour too!!!

**Comment by alumni** [comment]
Yes, I'm also struggling with productivity on MacOS even after some time of getting used to it. For this, Windows is still undoubtedly the king.

It would be really nice to have this feature since I'm currently stuck with the Mac.

**Comment by alumni** [comment]
@flying-sheep everything in my comment is written in first person, it was not generalized.

**Comment by alumni** [comment]
After learning from this issue that iTerm2 closes when its last window closes, I started checking if other apps have this too.

There's an app called Swift Quit that will close the app when there is only one window. Works well with VSCode - the last workspace will be restored when the app starts again (which was my main issue before).

**Comment by lllopo** [comment]
Just switched to MacOS and this drives me crazy. Hope the devs will take in consideration the failed #157263 and this one and will count the upvotes as enough to implement it.

**Comment by andrewmcdonald-oxb** [comment]
I'd also like to see this. It's a little annoying to have my dock cluttered with applications over time. I see there were 8 upvotes on the previous issue, and now 12 on this one. It's the same feature request - does that make the 20 needed (as per the auto-comment on the previous issue) for this to make it to the backlog?

(As an aside, that shows the problem with automatically closing issues in that circumstance. They might reach the needed engagement later down the line; it just takes time.)

**Comment by Saerujii** [comment]
> After learning from this issue that iTerm2 closes when its last window closes, I started checking if other apps have this too.
> 
> There's an app called Swift Quit that will close the app when there is only one window. Works well with VSCode - the last workspace will be restored when the app starts again (which was my main issue before).

Agreed. Annoying to have to open my workspace every time or pressing command + Q every time, because I normally work on Windows and when on mac I press the x button by habit.

**Comment by Gunnnn** [comment]
Please! =)

**Comment by alumni** [comment]
We should probably ask Apple TBH, they haven't improved their UX in years. An app should simply follow what the OS does and not come up with its own behavior.

I can't believe I'm going to say this, but after quite a long time working only with MacOS, I still miss the simplicity of Windows 😆

**Comment by lllopo** [comment]
Guys, just use Supercharge, works great.

**Comment by whyujjwal** [comment]
Hi — the Mac close-window vs quit behavior here (dock reopen → empty / wrong folder) is adjacent to what I’m validating for people who also use a layout/workspace stack.

Filter: you already use Moom, Workspaces, SnapState, Stay, EasySnaps, Spencer, Bunch, etc., and after dock or project switch you still lose the *correct* VS Code/Cursor project folder + terminal cwd (± browser tab)—not just geometry.

~45 min observe: last miss → your current tools → narrow prototype (Macfleet Project Context) with an honest restore report. Optional end ask: $29/year for that restore alone (yes / no / objection).

Reply or DM with tools you already use + a time window if interested.
